analyzers/src/SonarAnalyzer.Common/Rules/MethodsShouldNotHaveIdenticalImplementationsBase.cs

```diff
@@ -84,20 +83,15 @@ public abstract class MethodsShouldNotHaveIdenticalImplementationsBase<TMethodDe
                 },
                 SyntaxKinds);
 
+        protected virtual bool IsExcludedFromBeingExamined(ISymbol nodeContainingSymbol) =>
+            nodeContainingSymbol.Kind != SymbolKind.NamedType;
+
         protected static bool HaveSameParameters<TSyntax>(SeparatedSyntaxList<TSyntax>? leftParameters, SeparatedSyntaxList<TSyntax>? rightParameters)
-            where TSyntax : SyntaxNode
-        {
-            if (leftParameters == null && rightParameters == null)
-            {
-                return true;
-            }
-
-            if (leftParameters == null || rightParameters == null || leftParameters.Value.Count != rightParameters.Value.Count)
-            {
-                return false;
-            }
-
-            return leftParameters.Value.Zip(rightParameters.Value, (p1, p2) => new { p1, p2 }).All(tuple => tuple.p1.IsEquivalentTo(tuple.p2, false));
-        }
+            where TSyntax : SyntaxNode =>
+            (leftParameters == null && rightParameters == null)
+            || (leftParameters != null
+                && rightParameters != null
+                && leftParameters.Value.Count == rightParameters.Value.Count
+                && leftParameters.Value.Zip(rightParameters.Value, (p1, p2) => new { p1, p2 }).All(tuple => tuple.p1.IsEquivalentTo(tuple.p2)));
```

Comment:
I don't like `Zip` being missused.
```suggestion
                && leftParameters.Value.Select((left, index) => left.IsEquivalentTo(rightParameters.Value[index])).All(x => x));
```